BRANCHING.md

```diff
@@ -0,0 +1,66 @@
+  - So after Berkeley release: `berkeley` branch will become the new `master`. `berkeley` will be removed from `proof-systems`. `develop` will be renamed into `compatible`.
+
+
+
+### Diagram
+
+**TODO** fix diagram
+
+![git-flow_david-wong.png](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/09e92777-0232-401b-be44-9689d39ce22a/git-flow_david-wong.png)
+
+## Day to day: which branch should I use?
+
+When developing a feature, if it’s not something that breaks compatibility, then you should be developing a feature branch, called `foo_COMP` for example, based off of `compatible`.  If you’re not sure whether or not your changes are breaking, they probably are not and should build upon compatible.
+
+### Directions for merging your branch `foo_COMP`
+
+- There is a CI job called `merges cleanly to develop` which runs whenever you have a PR off of `compatible`.  If that CI job passes, then you can simply merge `foo_COMP` into `compatible` , and no further action is needed.
+- If `merges cleanly to develop` does NOT pass, then when you’re done with your changes to `foo_COMP` and the PR is all approved, make a new branch+PR based off of `foo_COMP` called `foo_DEVELOP` (for example), and then merge `develop` into `foo_DEVELOP`.  Fix any merge errors that result, then once `foo_DEVELOP` is approved, you can merge it into `develop`.
+- Once `foo_DEVELOP` is fully landed into `develop`, then go and manually re-run the `merges cleanly to develop` CI job in the original `foo_COMP` PR.  The CI job should now pass and go green.  You can now merge `foo_COMP`
+- If after making `foo_DEVELOP`, you need to make changes to `foo_COMP`, then make sure to merge the newly updated `foo_COMP` into `foo_DEVELOP`.  In order for the git magic to work, `foo_DEVELOP` needs to be a superset of the commits from `foo`, and it also needs to merge first.  You can make further changes post-merge in `foo_DEVELOP` as needed to ensure correctness
+
+### Releases
+
+(**TODO** is this outdated/needs to be removed?)
+
+`release/1.X.X` branches are made off of `compatible` and tagged with alpha and beta tags until the code is deemed stable, then the `release/1.X.X` branch is merged into `master` and given a stable tag. Whenever code is tagged, if anything is missing in in the upstream branches (compatible, develop) then the tagged branch is also merged back for consistency.
+
+`release/2.0.0`is the branch where Berkeley QA Net releases are being cut, between `compatible`and `develop` in the tree. So far nothing has been tagged there but there will be `2.0.0alphaX` tags once the code is more stable and we are closer to the Incentivized testnet.
+
+Unless it is an emergency, code should flow from feature branches into `compatible` then in batches into the release branch for tagging and testing
+
+# proof-systems Branching policy
```

Comment:
This should probably live in the proof-systems repo

Reply:
@mrmr1993 absolutely, but I thought it's better to discuss both at the same time here to avoid same discussion being in 2 PRs? And then I'll move it to `proof-systems`?

Otherwise I could move it now already.

Reply:
and we should have one for o1js too -- I'll ping the relevant folks